src/SongPro.js

```diff
@@ -74,33 +84,62 @@ export class SongPro {
 
     const line = new Line();
 
-    let captures = this.scan(text, /(\[[\w#b/]+])?([\w\s',.!()_\-"]*)/gi);
-    captures = flatten(captures);
-    const groups = chunk(captures, 2);
+    if (text.startsWith("|-")) {
+      line.tablature = text;
+    } else if (text.startsWith("| ")) {
+      let captures = this.scan(text, MEASURES_REGEX);
+      captures = flatten(captures);
 
-    for (let i = 0; i < groups.length; i++) {
-      const group = groups[i];
-      let chord = group[0];
-      let lyric = group[1];
+      const measures = [];
 
-      const part = new Part();
+      for (let i = 0; i < captures.length; i++) {
+        let chords = this.scan(captures[i], CHORDS_REGEX);
+        chords = flatten(chords);
 
-      if (chord) {
-        chord = chord.replace("[", "").replace("]", "");
+        const measure = new Measure();
+        measure.chords = chords;
+        measures.push(measure);
       }
 
-      if (chord === undefined) {
-        chord = "";
-      }
-      if (lyric === undefined) {
-        lyric = "";
-      }
+      line.measures = measures;
+    } else if (text.startsWith(">")) {
+      const matches = COMMENT_REGEX.exec(text);
 
-      part.chord = trim(chord);
-      part.lyric = trim(lyric);
+      if (matches == null) {
+        return;
+      }
 
-      if (!(part.chord === "" && part.lyric === "")) {
-        line.parts.push(part);
+      const comment = matches[1]; // strip!
+      line.comment = comment;
+    } else {
+      let captures = this.scan(text, CHORDS_AND_LYRICS_REGEX);
+      captures = flatten(captures);
+      const groups = chunk(captures, 2);
+
+      for (let i = 0; i < groups.length; i++) {
```

Comment:
any reason to go for indexed for-loop instead of iterator?

    for (const group of groups) {

thereby making the next line unnecessary.

See https://javascript.plainenglish.io/a-nice-way-to-iterate-over-collections-in-javascript-254b6f5d9907

Reply:
Probably not. Think I was just on autopilot. Thanks!

Reply:
Fixed!